Framework/Algorithms/src/Rebin2D.cpp

```diff
@@ -135,7 +135,7 @@ void Rebin2D::exec() {
     m_progress->report("Computing polygon intersections");
     const double vlo = oldYEdges[i];
     const double vhi = oldYEdges[i + 1];
-    for (size_t j = 0; j < numXBins; ++j) {
+    for (size_t j = 0; j < numXBins - 1; ++j) {
```

Comment:
I'd like to understand this a bit more. My memory is that there was an assumption here that `len(oldXEdges)==len(numXBins)+1` as X was always a histogram and it only made sense to rebin histograms. Does the input workspace have point values on X?

Reply:
Yeh I told Matthew that there are two scenarios - it seems that in the script in the issue (which is run in a doctest) the xedges have the same number of elements as the number of elements as the xbins (i.e. they are not edges but centres...) but not in the script from the documentation. He thought it might be a problem with the test, not the algorithm.

Reply:
do we need to handle both scenarios?

Reply:
Good question. We need in some way to handle it: whether we stop a workspace that is not binned from being allowed or we allow it and fix the loop. It feels odd to me that we would want to Rebin point data. I wonder if @gvardany could shed any light on the doctest as to whether the test should be used binned data on X and we should fix that and then have `Rebin2D` disallow point data?

Reply:
Can you clarify which doc-test we are talking about?

Reply:
I think @Mantid-Matthew knows which doctest as he created the original issue

Reply:
Looks like it's this: https://docs.mantidproject.org/nightly/algorithms/ComputeIncoherentDOS-v1.html

Reply:
Assuming it's this test, what I can say is that `stw` is histogram workspace. Why Rebin gets called over point data inside `ComputeIncoherentDOS`, I have no idea.

Reply:
OK thanks for looking into it. I think @DannyHindson has already solved this one - in his comment on issue #27162 he says he thinks it might be fixed by  "replacing two calls to inputWS->x(i) with inputWS->binEdges(i) in Rebin2D.cpp (L87) and FractionalBinning.cpp (L611)"

Reply:
Reading further, apparently there is a transposition happening inside `ComputeIncoherentDOS`, hence the point data. It looks like Rebin in mantid does accept point data, so Rebin2D should as well. So ideally the code should be made to work with both.

Reply:
OK thanks, I think Danny's proposed changes should do the job - I will test that now

Reply:
@gvardany That's fair. `binEdges` will just compute something based on the current spacing of the centers but that's better than crashing or not accepting it.